ggml-vulkan.cpp

```diff
@@ -4219,31 +4123,35 @@ bool ggml_vk_compute_forward(ggml_compute_params * params, ggml_tensor * tensor)
 
     GGML_ASSERT(extra->ready);
 
-    if (!extra->out_seqs.empty() || !extra->comp_seqs.empty()) {
-        // Do staging buffer copies
-        for (auto& cpy : extra->in_memcpys) {
-            memcpy(cpy.dst, cpy.src, cpy.n);
-        }
-        ggml_vk_submit(vk_device.transfer_queue, extra->in_seqs, VK_NULL_HANDLE);
-        if (extra->out_seqs.empty()) {
-            ggml_vk_submit(vk_device.compute_queue, extra->comp_seqs, vk_fence);
-        } else {
-            ggml_vk_submit(vk_device.compute_queue, extra->comp_seqs, VK_NULL_HANDLE);
-            ggml_vk_submit(vk_device.transfer_queue, extra->out_seqs, vk_fence);
-        }
+    // Do staging buffer copies
+    for (auto& cpy : extra->in_memcpys) {
+        memcpy(cpy.dst, cpy.src, cpy.n);
+    }
+    // Trigger staging events if any exists
+    if (extra->in0_staging_event != VK_NULL_HANDLE) {
+        vk_device.device.setEvent(extra->in0_staging_event);
+    }
+    if (extra->in1_staging_event != VK_NULL_HANDLE) {
+        vk_device.device.setEvent(extra->in1_staging_event);
+    }
```

Comment:
Please cast this as it's causing the compiler to error out.
```
ggml-vulkan.cpp:4131:34: error: ambiguous overload for ‘operator!=’ (operand types are ‘vk::Event’ and ‘std::nullptr_t’)
 4131 |     if (extra->in0_staging_event != VK_NULL_HANDLE) {
      |                                  ^
```
```
    if (extra->in0_staging_event != (vk::Event) VK_NULL_HANDLE) {
        vk_device.device.setEvent(extra->in0_staging_event);
    }
    if (extra->in1_staging_event != (vk::Event) VK_NULL_HANDLE) {
        vk_device.device.setEvent(extra->in1_staging_event);
    }
```

Reply:
Yeah, I noticed that (works on Arch but not on Ubuntu, for whatever reason) and forgot to upload the fix. Thank you.